### DISCUSSION TITLE
Workflows get starved and blocks workflow run of subsequent pushes

### DISCUSSION THREAD

**Discussion by du-it** [community/community#65882] [Programming Help]
### Body

I have a workflow chain each having this block:

```
concurrency:
  group: ${{ github.ref }}
  cancel-in-progress: true
```
 
 First it seems to do the job but today it happened that a called workflow was entered but then starved. 
 I am not bale to cancel this workflow, neither using `gh runs cancel <run-id>` nor using the cancel-button in the GHitHub GUI. 
 How can this happen and how can I solve the problem?
  
  Maybe of interest:
  The workflow chain is pipeline.yml --> wf_1.yml --> wf_2.yml
  In wf_1.yml the last job is:
```
    run_deploy_and_test_wf:
    needs: [build-be, build-fe]
    if: |
      !cancelled()
      && contains(needs.*.result, 'success')
      && !contains(needs.*.result, 'failure')
    uses: ./.github/workflows/wf_2.yml
    secrets: inherit
```
 The workflow starves directly after wf_2 was started, just telling me: 'Starting job...'

This warning also occurs: Failed to restore: The operation cannot be completed in timeout.
 
 What could be the reason and how can I stop this workflow?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by arafajs** [comment]
- The `concurrency` block with the same `group` value prevents concurrent runs of the workflow. So if a previous run is still in progress, a new run will wait indefinitely until the previous one finishes. This can cause the appearance of a stuck workflow.

- The workflow may be waiting on some external resource or process that is taking a long time, so it looks stuck but is still progressing slowly.

- There is a genuine bug or infinite loop in the workflow logic that causes it to become unresponsive.

Some things you could try to resolve this:

- Check the workflow run logs and jobs to see if they provide any clues on where it is getting stuck. 

- Try removing the `concurrency` block temporarily to see if that unblocks it.

- Reduce the workflow to bare minimum steps to isolate the problem.

- Set shorter timeouts on jobs.

- If it seems completely unresponsive, you may have to re-run the workflow to unblock it, making sure to change the `group` name to avoid concurrency issues.

- As a last resort, you can delete the entire workflow run history which will reset the state.

**Reply by du-it** [reply]
In the run log only 'Starting job...' appears.

As I read so far in the GitHub documentation [Using concurrency](https://docs.github.com/en/actions/using-jobs/using-concurrency)
```
concurrency:
  group: ${{ github.ref }}
  cancel-in-progress: true
```
should stop all pending as well as the running workflow. This is what I intend to do.

Currently I have set NO timeouts.

I do not have the option to re-run the workflow. There is only the 'Cancel workflow' button which causes no effect. Workflow status: Waiting

How can I delete the entire workflow history? I can't see any possibility.
